src/components/tx/TxSecurityWarnings/index.tsx

```diff
@@ -0,0 +1,88 @@
+import useAddressBook from '@/hooks/useAddressBook'
+import useAsync from '@/hooks/useAsync'
+import useSafeInfo from '@/hooks/useSafeInfo'
+import useWallet from '@/hooks/wallets/useWallet'
+import { useWeb3ReadOnly } from '@/hooks/wallets/web3'
+import { type RedefinedModuleResponse } from '@/services/security/modules/RedefineModule'
+import type { SecurityResponse } from '@/services/security/modules/types'
+import type { UnknownAddressModuleResponse } from '@/services/security/modules/UnknownAddressModule'
+import { dispatchTxScan, SecurityModuleNames } from '@/services/security/service'
+import CircularProgress from '@mui/material/CircularProgress'
+import type { SafeTransaction } from '@safe-global/safe-core-sdk-types'
+import { useMemo } from 'react'
+import { SecurityWarning, SecurityHint } from '../SecurityWarnings'
+
+const useUnknownAddress = (safeTransaction: SafeTransaction | undefined) => {
+  const addressBook = useAddressBook()
+  const addressBookAddresses = useMemo(() => Object.keys(addressBook), [addressBook])
+  const web3Provider = useWeb3ReadOnly()
+
+  return useAsync<SecurityResponse<UnknownAddressModuleResponse>>(() => {
+    if (!safeTransaction || !web3Provider) {
+      return
+    }
+
+    return dispatchTxScan({
+      type: SecurityModuleNames.UNKNOWN_ADDRESS,
+      request: {
+        knownAddresses: addressBookAddresses,
+        provider: web3Provider,
+        safeTransaction,
+      },
+    })
+  }, [safeTransaction, web3Provider, addressBookAddresses])
+}
+
+const useRedefine = (safeTransaction: SafeTransaction | undefined) => {
+  const { safe, safeAddress } = useSafeInfo()
+  const wallet = useWallet()
+
+  return useAsync<SecurityResponse<RedefinedModuleResponse>>(() => {
+    if (!safeTransaction || !wallet?.address) {
+      return
+    }
+    return dispatchTxScan({
+      type: SecurityModuleNames.REDEFINE,
+      request: {
+        chainId: Number(safe.chainId),
+        safeTransaction,
+        safeAddress,
+        walletAddress: wallet.address,
+        threshold: safe.threshold,
+      },
+    })
+  }, [safe.chainId, safe.threshold, safeAddress, safeTransaction, wallet?.address])
+}
```

Comment:
Are we going to wrap and dump every single module here? Can't we generalize this?